internal/state/module.go

```diff
@@ -452,8 +544,11 @@ func moduleCopyByPath(txn *memdb.Txn, path string) (*Module, error) {
 	return mod.Copy(), nil
 }
 
-func (s *ModuleStore) UpdateInstalledProviders(path string, pvs map[tfaddr.Provider]*version.Version) error {
+func (s *ModuleStore) UpdateInstalledProviders(path string, pvs map[tfaddr.Provider]*version.Version, pvErr error) error {
```

Comment:
I don't think we do, but I see that we aren't actually storing the error anywhere, so that needs fixing for sure! 😅 

The idea with most of these job "results" is that we persist both the result (if any, `nil` is fine too) and any error(s), giving downstream jobs the freedom to choose whether to take incomplete data and whether to run if the upstream has errored. This is important specifically in the context of https://github.com/hashicorp/terraform-ls/pull/1006